hack/wait-for-namespace.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+##############################################################################
+# wait-for-namespace.sh
+#
+##############################################################################
+set -e
+
+if [ $# -eq 0 ]
+  then
+    >&2 echo "No arguments supplied"
+    exit 1
+fi
+
+# process command line args
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    -n|--namespace)
+      shift;
+      NAMESPACES=( "$@" )
+      break;
+      ;;
+    -h|--help)
+      cat <<HELPMSG
+Valid command line arguments:
+  -n|--namespaces <name>: all of the namespaces we want to patch operator and wait for
+  -h|--help : this message
+HELPMSG
+      exit 1
+      ;;
+    *)
+      echo "Unknown argument [$key]. Aborting."
+      exit 1
+      ;;
+  esac
+done
+
+IS_MAISTRA=$([ "$(oc get crd | grep servicemesh | wc -l)" -gt "0" ] && echo "true" || echo "false")
+
+if [ "${IS_MAISTRA}" == "false" ]; then
+  KIALI_CR_NAMESPACE_NAME="$(oc get kiali --all-namespaces -o custom-columns=NS:.metadata.namespace,N:.metadata.name --no-headers | sed 's/  */:/g')"
+  KIALI_CR_NAMESPACE="$(echo ${KIALI_CR_NAMESPACE_NAME} | cut -d: -f1)"
+  KIALI_CR_NAME="$(echo ${KIALI_CR_NAMESPACE_NAME} | cut -d: -f2)"
+  ACCESSIBLE_NAMESPACES="$(oc get kiali $KIALI_CR_NAME -n $KIALI_CR_NAMESPACE -o jsonpath='{.spec.deployment.accessible_namespaces}')"
+
+  # All namespaces are accessible, no need to add namespaces access to CRD
+  if [ "${ACCESSIBLE_NAMESPACES}" != "**" ]; then
+    # accessible_namespaces field is not defined in the CRD, initializing value to empty array
+    if [ "${ACCESSIBLE_NAMESPACES}" == "" ]; then
+      oc patch kiali $KIALI_CR_NAME -n $KIALI_CR_NAMESPACE --type=merge -p '{"spec": {"deployment": {"accessible_namespaces": []}}}'
+    fi
+
+    for NAMESPACE in ${NAMESPACES[@]}; do
+      oc patch kiali $KIALI_CR_NAME -n $KIALI_CR_NAMESPACE --type=json '-p=[{"op": "add", "path": "/spec/deployment/accessible_namespaces/-", "value":"'$NAMESPACE'"}]'
+    done
+
+    echo -n "Waiting for operator to finish reconciling the CR named [$KIALI_CR_NAME] located in namespace [$KIALI_CR_NAMESPACE]"
+    while [ "$KIALI_CR_REASON" != "Successful" -o "$KIALI_CR_STATUS" != "True" ]; do
+      sleep 1
+      echo -n "."
+      KIALI_CR_REASON="$(oc get kiali $KIALI_CR_NAME -n $KIALI_CR_NAMESPACE -o jsonpath='{.status.conditions[?(@.message=="Awaiting next reconciliation")].reason}')"
+      KIALI_CR_STATUS="$(oc get kiali $KIALI_CR_NAME -n $KIALI_CR_NAMESPACE -o jsonpath='{.status.conditions[?(@.message=="Awaiting next reconciliation")].status}')"
+    done
+    echo
+    echo "Done reconciling"
+  fi
+fi
+
+for NAMESPACE in ${NAMESPACES[@]}; do
+  oc wait --for=condition=Ready pods --all -n "$NAMESPACE" --timeout 60s || true
+  oc wait --for=condition=Ready pods --all -n "$NAMESPACE" --timeout 60s
```

Comment:
Don't understand why wait is done twice here (?).

Reply:
Not sure, this part is copied from `wait-for-namespace.sh` script in Kiali repo, made by @matejnesuta in https://www.github.com/kiali/kiali/pull/6257. Do you remember why wait is done twice here? Probably there is a good reason to do that.

Reply:
Hello Fernando. The code in the script originated from one of our Jenkins jobs, which is used for installing the demoapps. I had to re-use this piece of code in my Cypress hook, otherwise it would not work on our OCP pipelines, so I decided to create this little script and it is now executed both by the hook and by the Jenkins pipeline. 

Frankly, I am also not sure why the line is present twice here.

Reply:
Ok, in that case I will remove the `oc wait` command duplicated

Reply:
I see this code at the end of this script:

```bash
for NAMESPACE in ${NAMESPACES[@]}; do
  oc wait --for=condition=Ready pods --all -n "$NAMESPACE" --timeout 60s
done

sleep 80
```

It looks like this is simply waiting for everything to start (i.e. it waits for all pods in each namespace to be ready, then it waits another 80 seconds before exiting).

Notice the line `oc wait` has a timeout of 60s. If 60 seconds passes and all the pods are not ready, the `oc wait` will exit with a non-zero exit code. This will abort the script (due to `set -e` on line 6, which tells the script to exit on any command that results in an error). That means it will not wait for any more namespaces to be ready and it will not wait that extra 80 seconds.

Now, notice the line you removed in the [last commit](https://github.com/kiali/openshift-servicemesh-plugin/pull/245/commits/8a92d07ebfa45ea085994dbb05a4771714460a9b) had `|| true` at the end - this means even if the `oc wait` timed out, it will not cause the script to abort. So whoever wrote that code originally must have wanted that code to keep waiting even if one namespace isn't ready (i.e. he didn't want the script to abort if `oc wait` timed out) - he wanted to wait for the other namespaces and that extra 80 seconds even if one namespace wasn't ready after 60s. I guess this is to wait as long as possible so the test components have a chance to start fully before the tests begin to run.

I will leave it up to you if you want to maintain that same behavior as before and put back that `|| true` or if you want to leave it out. I'll approve this PR in case you want to leave it as-is and you can merge.

Reply:
Good catch @jmazzitelli. Eventually this script is used for cypress test configuration, so I prefer to set `|| true` and don't abort the test process if the pods are not ready (to avoid failures if the system is quite slow that day). If they are not ready after the extra 80s wait then cypress tests will fail afterwards. @matejnesuta what do you think?

Reply:
I have no problems removing one of the `oc wait` commands as long as it works.